lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -274,8 +274,8 @@ function _campaign_resource_signup($nid, $values) {
     unset($values['northstar_id']);
   }
 
-  // If we pass either a Northstar ID or Drupal UID to `uid`, convert it to a real UID.
-  $values['uid'] = dosomething_user_convert_to_legacy_id($values['uid']);
+  // If a Drupal UID is passed to `uid`, convert it to the user's Northstar ID.
+  $values['uid'] = dosomething_user_convert_to_northstar_id($values['uid']);
```

Comment:
> @DFurnes Since we have the `northstar_id` of the user (assuming everyone has it), would it work to send the `northstar_id` directly to [`dosomething_rogue_send_signup_to_rogue`](https://github.com/DoSomething/phoenix/blob/6e80ae77499c1fd8b30508c1bad7ab6e8dd6d192/lib/modules/dosomething/dosomething_signup/dosomething_signup.module#L286) and update this to take that instead of `account`?

I think that's a great idea! 👍

> We can also delete [this](https://github.com/DoSomething/phoenix/blob/6e80ae77499c1fd8b30508c1bad7ab6e8dd6d192/lib/modules/dosomething/dosomething_signup/dosomething_signup.module#L279-L281) since Rogue won't create a new Signup if one already exists. 

Let's make sure that Phoenix returns the same response (the signup ID) if we do.